modules/ROOT/pages/live-booting-ipxe.adoc

```diff
@@ -0,0 +1,57 @@
+
+NOTE: The live system requires at least 3 GiB of RAM.
+
+== PXE images
+
+NOTE: Before August 2020, the Fedora CoreOS PXE image included two components: a `kernel` image and an `initramfs` image.  Beginning August 11, a third `rootfs` image will be added.  During an initial migration period, the `rootfs` image will be optional, and Fedora CoreOS will display a warning on login if the system was booted without it.  After the migration period, the `rootfs` image will be mandatory and the live PXE system will not boot without it.
+
+The migration timeline is:
+
+- August 11: `rootfs` image available in `next`, `testing`, and `stable` streams
+- August 25: `rootfs` image required in `next` stream
+- September 22: `rootfs` image required in `testing` stream
+- October 6: `rootfs` image required in `stable` stream
+
+To boot with the `rootfs` artifact, make one of the following changes to your PXE config:
+
+- Specify both `initramfs` and `rootfs` files as initrds in your PXE configuration.  In PXELINUX, put both file paths in the `initrd` directive, separated by commas, or specify two `initrd=` parameters in your `append` line.
+- Concatenate the `initramfs` and `rootfs` files together, and specify the combined file as the initrd.
+- Specify only the `initramfs` file as the initrd, and pass an HTTP(S) URL for the `rootfs` using the `coreos.live.rootfs_url=` kernel argument.
+
+For more information on this change, see the https://github.com/coreos/fedora-coreos-tracker/issues/390[tracker issue].
+
+== Setting up the Boot Script
+
+An iPXE-capable machine needs to be provided with a relevant Boot Script to fetch and load FCOS artifacts.
+
+The example below shows how to load those directly from Fedora infrastructure. For performance and reliability reasons it is recommended to mirror them on the local infrastructure, and then tweak the `BASEURL` as needed.
+
+[source]
+----
+#!ipxe
+
+set STREAM stable
+set VERSION 32.20200726.3.0
+set CONFIGURL https://example.com/config.ign
+
+set BASEURL https://builds.coreos.fedoraproject.org/prod/streams/${STREAM}/builds/${VERSION}/x86_64
+
+kernel ${BASEURL}/fedora-coreos-${VERSION}-live-kernel-x86_64 ignition.firstboot ignition.platform.id=metal ignition.config.url=${CONFIGURL}
+initrd ${BASEURL}/fedora-coreos-${VERSION}-live-initramfs.x86_64.img
+initrd ${BASEURL}/fedora-coreos-${VERSION}-live-rootfs.x86_64.img
+
+boot
+----
+
+== Update process
+
+Since the traditional FCOS upgrade process requires a disk, live-PXE systems are not able to auto-update in place. For this reason, Zincati is not running there.
+
+Instead, it is reccomended that images references in the PXE configuration are regularly refreshed. Once infrastructure and configurations are updated, the live-PXE instance simply needs to be rebooted in order to start running the new FCOS version.
```

Comment:
Is there any way in the ipxe environment/config to do some logic to figure out what the latest is and use that? That would allow people to follow the latest without having to update their configs. The flip side of that is that someone could automate generating the ipxe config and just substitute in the latest version.

Reply:
Not that I've seen. Plus I'm not sure that would be a very good approach. We somehow consciously decided to steer away from the Container Linux way (having a `current` directory and not encoding version into file name), and this mostly follows from it.

What we could add here as a followup is showing how to mirror the current tip of a stream via `coreos-installer download`. That should make it slightly easier to automate and refresh (not fully though).